primitives/consensus/beefy/src/lib.rs

```diff
@@ -251,6 +251,41 @@ impl<Number, Id, Signature> EquivocationProof<Number, Id, Signature> {
 	}
 }
 
+/// Proof of authority misbehavior on a given set id.
+/// This proof shows commitment signed on a different fork.
+#[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
+pub struct ForkEquivocationProof<Number, Id, Signature, Header> {
+	/// Commitment for a block on different fork than one at the same height in
+	/// this client's chain.
+	/// TODO: maybe replace {commitment, signatories} with SignedCommitment
+	/// (tradeoff: SignedCommitment not ideal since sigs optional, but fewer
+	/// types to juggle around) - check once usage pattern is clear
+	pub commitment: Commitment<Number>,
+	/// Signatures on this block
+	/// TODO: maybe change to HashMap - check once usage pattern is clear
+	pub signatories: Vec<(Id, Signature)>,
+	/// The proof is valid if
+	/// 1. the header is in our chain
+	/// 2. its digest's payload != commitment.payload
+	/// 3. commitment is signed by signatories
+	pub correct_header: Header,
```

Comment:
I think we should just go with the mmr historic root membership proofs from the get-go instead of this `correct_header` temporary thing.

This one is not really useful as the reporting window is only ~7hours before `correct_header` cannot be verified on-chain anymore because of headers map pruning. Colluding validators can simply stop producing BEEFY finality for 7 hours then provide "valid" BEEFY proofs but for invalid forks to the bridge without being slashed.

And doing it like this in two steps, first using `correct_header`, then `payload_membership_proof` means dealing with breaking API changes, bumping API versions, etc.

I think it's not worth it.